compiler-core/src/analyse/imports.rs

```diff
@@ -242,28 +224,53 @@ impl<'a> Importer<'a> {
         Ok(())
     }
 
-    fn register_module_usage(&mut self, import: &Import<()>) {
-        if import.unqualified_types.is_empty() && import.unqualified_values.is_empty() {
-            // When the module has no unqualified imports, we track its usage
-            // so we can warn if not used by the end of the type checking
-            let _ = self
-                .environment
-                .unused_modules
-                .insert(import.used_name(), import.location);
-        } else if let Some(as_name) = &import.as_name {
-            // When the module has a name, we also track its as_name usage
-            // so we can warn if not used by the end of the type checking
-            let _ = self
-                .environment
-                .unused_modules
-                .insert(as_name.name.clone(), as_name.location);
-            // We also register it's name to differentiate between unused module
-            // and unused module name. See 'convert_unused_to_warnings'.
+    fn register_module(
+        &mut self,
+        import: &Import<()>,
+        import_info: &'a ModuleInterface,
+    ) -> Result<(), Error> {
+        let (used_name, location, aliased) = match import.as_name.as_ref() {
```

Comment:
Please use `used_name` here instead of pattern matching on `as_name` as detailed here:

https://github.com/gleam-lang/gleam/pull/2362#discussion_r1367033440